### PR TITLE
As a user, I want to be able to search CloudElements folders/files

### PR DESCRIPTION
**Pull request by vetostream**
This allows users to search through files/folders on an element they are accessing.

### CODE REVIEW

File: divvy/cloudfilebrowser.js [hunk 1]
```diff
@@ -720,6 +734,56 @@ var cloudFileBrowser = (function() {
...
 
+        performSearch: function(keywoard, element, path) {
+            var callbackArgs = {
+                'element' : element,
+                'path' : '/'
+            };
+
+            cloudFileBrowser.showLoading();
+
+            if(keywoard === '' || keywoard === undefined) {
```
Comment (nmallare) [inline divvy/cloudfilebrowser.js]: keyword*

Comment (nmallare) [reply divvy/cloudfilebrowser.js]: Also, please add some comments explaining the difference between no keyword and keyword. Also, what's going to be the most common case? We should add that case first.

File: divvy/cloudfilebrowser.js [hunk 2]
```diff
@@ -720,6 +734,56 @@ var cloudFileBrowser = (function() {
...
+
+        bindSearchBox: function(element, path) {
+            var self = this;
+            $('#js-search-open').on('click', function() {
+                $('#js-search-box-form-wrapper').show();
+                self.disableSearchOpen();
+            });
+
+            $('#js-search-box').on('keypress', function(event) {
+                if(event.keyCode === 13) {
```
Comment (nmallare) [inline divvy/cloudfilebrowser.js:787]: Add a comment so the next person knows that `13 = enter`.

File: divvy/cloudfilebrowser.js [hunk 3]
```diff
@@ -911,35 +975,52 @@ var cloudFileBrowser = (function() {
...
         buildTable: function(data, isNew, path, element) {
-
             if (isNew == true) {
 
                 var tableHTML = '',
                     trailingpath;
 
                 cloudFileBrowser.selectedFiles[element] = new Array();
 
+                tableHTML += '<div class="search-wrapper">' +
```
Comment (nmallare) [inline divvy/cloudfilebrowser.js:1029]: Doesn't this code make you appreciate Divvy 3.0 frontend? :)

File: divvy/cloudfilebrowser.scss [hunk 4]
```diff
@@ -405,6 +405,44 @@ the License.
     }
 
     #services-containers > div.provisioned > a.provision { display: none }
+
+    #services-containers > div.provisioned > div.search-wrapper {
```
Comment (nmallare) [inline divvy/cloudfilebrowser.scss:409]: Without looking too much into this CSS, I just want to make sure you reused as much of the styling as we already have in Divvy 3.0.

File: divvy/cloudfilebrowser.js [hunk 5]
```diff
@@ -466,6 +470,16 @@ var server = (function() {
                 this.authHeader(CloudElements.getUTkn(), CloudElements.getOTkn(), tkn), params, cb, cbArgs);
         },
 
+        search: function(tkn, path, keywoard, cb, cbArgs) {
+            var params = {
+                'path': path,
+                'text': keywoard,
+            }
+
+            _server.call('api-v2/hubs/documents/search', 'Get',
```
Comment (nmallare) [inline divvy/cloudfilebrowser.js:479]: Is `Get` correct here? Should it be `get` or `GET`, instead?

**Review by nmallare — DISMISSED** [review]
Overall, really solid! Great work :+1:. I'm going to deploy this code locally and provide some additional feedback re: styling.

File: divvy/cloudfilebrowser.js (hunk 5 above)
Comment (vetostream) [reply divvy/cloudfilebrowser.js:479]: @nmallare `Get` is used all through out the code, should I change those too?

File: divvy/cloudfilebrowser.js (hunk 3 above)
Comment (vetostream) [reply divvy/cloudfilebrowser.js:1029]: Absolutely!

File: divvy/cloudfilebrowser.scss (hunk 4 above)
Comment (vetostream) [reply divvy/cloudfilebrowser.scss:409]: Reused sir.

**Comment by nmallare** [comment]
@vetostream Here are a few things I've noticed:

- When I searched with an apostrophe `'`, it failed with a red box that just said "error"
- Can we add some padding to the left of the `(x)` so that the text doesn't hide behind it?
image
- Let's go ahead and widen the search input to 225px
- Upon further inspection, the CSS for the search bar seems a little funky. Please check either the header bar of the app or the search input on the global search results page (after making a global search using the header; you cannot do this in your local environment).

**Comment by vetostream** [comment]
@nmallare all fixed sir.

**Comment by nmallare** [comment]
@vetostream Perfecto. Go ahead and open up a PR in the DivvyHQ repository using your last commit has as the cloud-file-browser library version. Once Tony approves, you can merge this code and then I can show you how to version. And then we can update your other PR with the new version and merge it. Make sense?

**Review by nmallare — APPROVED** [review]

**Comment by nmallare** [comment]
@vetostream Please see these comments, https://github.com/DivvyHQ/divvyhq/pull/2595#issuecomment-491012210.

**Pull request merged**